test/unittest/component/test_listener.py

```diff
@@ -172,3 +189,53 @@ def test_listener_cleanup(db, data):
 
     db.remove('listener', listener1.identifier, force=True)
     assert not db.databackend.check_output_dest(listener1.predict_id)
+
+
+def test_listener_chaining_with_trainer(db):
+    db.cfg.auto_schema = True
+    table = db['test']
+
+    def insert_random(start=0):
+        data = []
+        for i in range(5):
+            y = int(random.random() > 0.5)
+            data.append(
+                Document(
+                    {
+                        "x": i + start,
+                        "y": y,
+                    }
+                )
+            )
+
+        db.execute(table.insert(data))
+
+    # Insert data
```

Comment:
Please add comments - why is this a useful test? Can we now solve the problem mentioned on the issue (https://github.com/orgs/superduper-io/projects/1?pane=issue&itemId=71398398)?